### PR TITLE
Feat/auth0 access

### PR DESCRIPTION
**Pull request by PascalUlor**
#### What does this PR do?

Implement social authentication from frontend
add jwt-decode file in utils
rename all rendered files and views to `.jsx` extensions

#### Description of Task to be completed?
write tests and add mobile responsiveness

#### Any background context you want to provide?

- [ ] I have performed a self-review of my own code
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works .
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/views/SignupPage.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import queryString from 'query-string';
 const SignupPage = (props) => {
   const isAuth = isLoggedIn();
   const dispatch = useDispatch()
+
```
Comment (attaradev) [inline src/components/views/SignupPage.jsx]: I think we can use the `useLocation` hook from `react-router-dom` and not pass it as props

File: src/utils/auth.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  const { currentUser } = JSON.parse(localStorage.getItem('persist:hackton'));
+  const { token } = JSON.parse(currentUser);
+  if (token) {
+    const decodedToken = jwtDecode(token);
+
+    const currentTime = Date.now() / 1000;
+
+    if (decodedToken.exp < currentTime) {
+      // Logout user
+      return false;
```
Comment (attaradev) [inline src/utils/auth.js:13]: Can we create a `useAuth` hook for this, to avoid trying to fetch the information from localStorage?

**Review by attaradev — COMMENTED** [review]
Great work on the auth0 implementation. I have left some comments in some files.

File: src/components/views/SignupPage.jsx (hunk 1 above)
Comment (PascalUlor) [reply src/components/views/SignupPage.jsx]: NIce! Done

File: src/utils/auth.js (hunk 2 above)
Comment (PascalUlor) [reply src/utils/auth.js:13]: ok. But I think we can roll this out for our RC1 approval and refactor afterwards

**Pull request merged**